src/SplitIO/Component/Cache/SplitCache.php

```diff
@@ -77,6 +84,20 @@ public function getSplitNames()
         return array_map('self::getSplitNameFromCacheKey', $splitKeys);
     }
 
+    /**
+     * @param array(string) List of flag set names
+     * @return array(string) List of all feature flag names by flag sets
+     */
+    public function getNamesByFlagSets($flagSets)
+    {
+        $cache = Di::getCache();
+        $toReturn = array();
+        foreach ($flagSets as $flagSet) {
```

Comment:
Can `$flagSets` be `null`?
```suggestion
        if (is_array($flagSets)) {
            foreach ($flagSets as $flagSet) {
```

Reply:
at this level, input validation should pre-check nullable or other type that's not array